libafl/src/generators/mod.rs

```diff
@@ -55,31 +48,21 @@ where
     }
 }
 
-impl<R> RandBytesGenerator<R>
-where
-    R: Rand,
-{
+impl RandBytesGenerator {
     /// Returns a new [`RandBytesGenerator`], generating up to `max_size` random bytes.
     #[must_use]
     pub fn new(max_size: usize) -> Self {
-        Self {
-            max_size,
-            phantom: PhantomData,
-        }
+        Self { max_size }
     }
 }
 
 #[derive(Clone, Debug)]
 /// Generates random printable characters
-pub struct RandPrintablesGenerator<R> {
+pub struct RandPrintablesGenerator {
     max_size: usize,
-    phantom: PhantomData<R>,
 }
 
-impl<R> Generator<BytesInput, R> for RandPrintablesGenerator<R>
-where
-    R: Rand,
-{
+impl<R: Rand> Generator<BytesInput, R> for RandPrintablesGenerator {
```

Comment:
Any reason not to leave this as a `where` clause? I think that's our general code style so far